lib/matplotlib/tests/test_mlab.py

```diff
@@ -1388,32 +1296,37 @@ def createStim(self, fstims, iscomplex, sides, nsides, len_x=None,
         if iscomplex:
             y = y.astype('complex')
 
-        self.Fs = Fs
-        self.sides = sides
-        self.fstims = fstims
+        # Interestingly, the instance on which this fixture is called is not
+        # the same as the one on which a test is run. So we need to modify the
+        # class itself when using a class-scoped fixture.
```

Comment:
I think the pytesty-way to do this would be to return all these values from the fixture (e.g. as a tuple), and let the test functions take `stim` as fixture arg?  As it's class-scoped it'll be only computed once.